orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/ecs/EcsServerGroupCreator.groovy

```diff
@@ -86,6 +129,47 @@ class EcsServerGroupCreator implements ServerGroupCreator, DeploymentDetailsAwar
     return [[(ServerGroupCreator.OPERATION): operation]]
   }
 
+  private Supplier<Iterable<Object>> fetchAndParseArtifact(Artifact artifact) {
+    return { ->
+      Response artifactText = oortService.fetchArtifact(artifact)
+      try {
+        if(artifactText != null && artifactText.getBody() != null){
+          return yamlParser.get().loadAll(artifactText.getBody().in());
+        } else{
+          throw new ConfigurationException("Invalid artifact configuration or task definition artifact is null")
+        }
+      } catch (Exception e) {
+        throw new IllegalStateException(e);
+      }
+    }
+  }
+
+  private List<Map<Object, Object>> coerceArtifactToList(Object artifact) {
+    Map<Object, Object> singleArtifact =
+        objectMapper.convertValue(artifact, new TypeReference<Map<Object, Object>>() {})
+    return List.of(singleArtifact)
+  }
+
+  private Map<Object, Object> getSpelEvaluatedArtifact(
+      List<Map<Object, Object>> unevaluatedArtifact, StageExecution stage) {
+    Map<String, Object> processorInput = ImmutableMap.of("artifact", unevaluatedArtifact);
```

Comment:
Is there a reason `unevaluatedArtifact` is still a list here? does `contextParameterProcessor.process()` need to take in a map of string:list?

Reply:
discussed offline: `StreamSupport.stream(...)` returns a list and its usage here matches its use in similar logic for other providers.